src/benchmarks/micro/corefx/System.Buffers/Base64EncodeDecodeTests.cs

```diff
@@ -0,0 +1,117 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using BenchmarkDotNet.Attributes;
+using BenchmarkDotNet.Extensions;
+using MicroBenchmarks;
+
+namespace System.Buffers.Text.Tests
+{
+    [BenchmarkCategory(Categories.CoreFX)]
+    public class Base64EncodeDecodeTests
+    {
+        [Params(1000)]
+        public int NumberOfBytes { get; set; }
+
+        private byte[] _decodedBytes;
+        private byte[] _encodedBytes;
+
+        private char[] _encodedChars;
+
+        [GlobalSetup(Target = nameof(Base64Encode))]
+        public void SetupBase64Encode()
+        {
+            _decodedBytes = ValuesGenerator.Array<byte>(NumberOfBytes);
+            _encodedBytes = new byte[Base64.GetMaxEncodedToUtf8Length(NumberOfBytes)];
+        }
+
+        [Benchmark]
+        public OperationStatus Base64Encode() => Base64.EncodeToUtf8(_decodedBytes, _encodedBytes, out _, out _);
+
+        [GlobalSetup(Target = nameof(Base64EncodeDestinationTooSmall))]
+        public void SetupBase64EncodeDestinationTooSmall()
+        {
+            _decodedBytes = ValuesGenerator.Array<byte>(NumberOfBytes);
+            _encodedBytes = new byte[Base64.GetMaxEncodedToUtf8Length(NumberOfBytes) - 1]; // -1
+        }
+
+        [Benchmark]
+        public OperationStatus Base64EncodeDestinationTooSmall() => Base64.EncodeToUtf8(_decodedBytes, _encodedBytes, out _, out _);
+
+        [GlobalSetup(Target = nameof(Base64EncodeBaseline))]
+        public void SetupBase64EncodeBaseline()
+        {
+            _decodedBytes = ValuesGenerator.Array<byte>(NumberOfBytes);
+            _encodedChars = new char[Base64.GetMaxEncodedToUtf8Length(NumberOfBytes)];
+        }
+
+        [Benchmark]
+        public int Base64EncodeBaseline() => Convert.ToBase64CharArray(_decodedBytes, 0, _decodedBytes.Length, _encodedChars, 0);
+
+        [GlobalSetup(Target = nameof(Base64Decode))]
+        public void SetupBase64Decode()
+        {
+            _encodedBytes = ValuesGenerator.Array<byte>(NumberOfBytes);
+            _decodedBytes = new byte[Base64.GetMaxEncodedToUtf8Length(NumberOfBytes)];
+        }
+
+        [Benchmark]
+        public OperationStatus Base64Decode() => Base64.DecodeFromUtf8(_encodedBytes, _decodedBytes, out _, out _);
```

Comment:
Same question... why is this called Base64Decode rather than Base64DecodeFromUtf8, or something like that?